.github/workflows/create-release.yml

```diff
@@ -0,0 +1,98 @@
+---
+name: Create GitHub release
+
+on:
+  push:
+    tags:
+      - '*v[0-9]*'
+
+jobs:
+  build-release-assets:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-14, macos-12]
+        arch: [amd64, arm64]
+        python-version: ['3.9', '3.10', '3.11', '3.12']
+        exclude:
+          - os: windows-latest
+            arch: arm64  # Windows doesn't support arm64 in this context
+          - os: macos-12
+            arch: amd64  # macOS 11 should be used for arm64 (Apple Silicon)
+          - os: macos-14
+            arch: amd64  # macOS 11 should be used for arm64 (Apple Silicon)
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |-
+          python -m pip install --upgrade pip
+          pip install build
+
+      - name: Build python wheels + package source code
+        run: python -m build
+
+      - name: Upload builds
+        uses: actions/upload-artifact@v3
+        with:
+          name: python-package-${{ matrix.python-version }}
+          path: dist/*
+
+  create-github-release:
+    needs: build-release-assets
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Get release version from "czespressif/__init__.py"
+        id: get_version
+        shell: bash
+        run: |-
+          version=$(grep '__version__' czespressif/__init__.py | sed -E "s/__version__ = '([^ ]*)'.*/\1/")
+          echo "VERSION=$version" >> $GITHUB_OUTPUT
+
+      - name: Install dependencies
+        run: python -m pip install pip czespressif --upgrade
```

Comment:
At this point, you are installing dependencies from the previous version.
I believe it should be `python -m pip install pip . --upgrade`